src/modules/comments/comments.service.ts

```diff
@@ -0,0 +1,103 @@
+import { NotFoundException } from './../../shared/exceptions/http.exception';
+import { Repository } from 'typeorm';
+import {
+  BadRequestException,
+  InternalServerErrorException,
+} from '../../shared/exceptions/http.exception';
+import { Comment } from './comments.entity';
+import { CreateCommentDto } from './dto/create-comment.dto';
+import { UpdateCommentDto } from './dto/update-comment.dto';
+import { v4 as uuidv4 } from 'uuid';
+import { User } from '../user/user.entity';
+import { Blog } from '../blog/blog.entity';
+
+export class CommentService {
+  constructor(
+    private commentRepository: Repository<Comment>,
+    private userRepository: Repository<User>,
+    private blogRepository: Repository<Blog>
+  ) {}
+
+  async createComment(blogId:string, createCommentDto: CreateCommentDto) {
+    try {
+      const { userId, content } = createCommentDto;
+
+      const author = await this.userRepository.findOne({ where: { id: userId } });
```

Comment:
if this is comment author, you can create one middleware to check user validness, so you don't need to check comment author in service as middleware won't pass request to service if user is invalid

Reply:
Ok, I'll remove this code from here and connect my guards